From f12b179950fe4bf73828ce5febc5ec5d80301c52 Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Tue, 22 May 2018 16:17:32 -0700
Subject: [PATCH] adjust for API changes in LLVM

addPassesToEmitFile has an extra parameter now for DWO output.  Adjust the call
accordingly.
---
 src/LLVM_Output.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/LLVM_Output.cpp b/src/LLVM_Output.cpp
index b6867e052e2..adbb34fcf34 100644
--- a/src/LLVM_Output.cpp
+++ b/src/LLVM_Output.cpp
@@ -378,7 +378,11 @@ void emit_file(const llvm::Module &module_in, Internal::LLVMOStream& out, llvm::
     target_machine->Options.MCOptions.AsmVerbose = true;
 
     // Ask the target to add backend passes as necessary.
+#if LLVM_VERSION < 70
     target_machine->addPassesToEmitFile(pass_manager, out, file_type);
+#else
+    target_machine->addPassesToEmitFile(pass_manager, out, nullptr, file_type);
+#endif
 
     pass_manager.run(*module);
 }
